continuous_integration/environment-mindeps-optional.yaml

```diff
@@ -0,0 +1,56 @@
+name: test-environment
+channels:
+  - conda-forge
+dependencies:
+  # required dependencies
+  - packaging=20.0
+  - python=3.8
+  - pyyaml=5.3.1
+  - click=8.0  # min version required by distributed
+  - cloudpickle=1.5.0  # min version required by distributed
+  - partd=1.2.0
+  - fsspec=2021.09.0
+  - importlib-metadata=4.13.0
+  - toolz=0.10.0  # min version required by distributed
```

Comment:
Worth noting here that `toolz=0.10.0` is also the minimum version required to use `cytoolz` - not sure if that motivates a bump of the minimum version for Dask